cache/cache.go

```diff
@@ -257,13 +299,38 @@ func (c *Cache) validateWatch(key string, op clientv3.Op) (pred KeyPredicate, er
 	startKey := []byte(key)
 	endKey := op.RangeBytes() // nil = single key, {0}=FromKey, else explicit range
 
-	if err := c.validateWatchRange(startKey, endKey); err != nil {
+	if err := c.validateRequestRange(startKey, endKey); err != nil {
 		return nil, err
 	}
 	return KeyPredForRange(startKey, endKey), nil
 }
 
-func (c *Cache) validateWatchRange(startKey, endKey []byte) error {
+func (c *Cache) validateGet(key string, op clientv3.Op) (KeyPredicate, error) {
+	if op.IsCountOnly() ||
+		op.IsPrevKV() ||
+		op.IsSortSet() ||
+		op.Limit() != 0 ||
+		op.MinModRev() != 0 ||
+		op.MaxModRev() != 0 ||
+		op.MinCreateRev() != 0 ||
+		op.MaxCreateRev() != 0 ||
+		// cache now only serves serializable reads of the latest revision (rev == 0).
+		op.Rev() != 0 ||
+		!op.IsSerializable() {
+		return nil, ErrUnsupportedRequest
+	}
```

Comment:
This check if getting heavy. In a follow up, let's refactor this to something like:
```go
switch {
    case op.IsCountOnly():
        return nil, fmt.Errorf("%w: CountOnly() not supported", ErrUnsupportedRequest)
    case op.IsPrevKV():
        return nil, fmt.Errorf("%w: PrevKV() not supported", ErrUnsupportedRequest)
    ...
}
```
We should get some insight into what option was set and not supported.